### PR TITLE
PHPStan Generics

### PR DESCRIPTION
**Pull request by hrach**
Related PHPStan issues:
- [ ] phpstan/phpstan#2928 - Generic bound (actually, it may not be needed in the end)
- [x] phpstan/phpstan#2929 - Assign & array-shape
- [x] phpstan/phpstan#3263 - Template bound is lost in get_class
- [ ] phpstan/phpstan#3290 - Star projection

### CODE REVIEW

File: src/Model/IModel.php [hunk 1]
```diff
@@ -22,25 +22,35 @@ public function hasRepositoryByName(string $name): bool;
 
 	/**
 	 * Returns repository by repository name.
+	 * @phpstan-template T of IRepository<IEntity>
+	 * @phpstan-return T
```
[the remark this answers is not in the document]
Comment (hrach) [reply src/Model/IModel.php:26]: Thanks!

File: src/Model/SimpleRepositoryLoader.php [hunk 2]
```diff
@@ -14,12 +14,15 @@
...
+	/**
+	 * @var IRepository[]
+	 * @phpstan-var array<class-string<IRepository<\Nextras\Orm\Entity\IEntity>>, IRepository<\Nextras\Orm\Entity\IEntity>>
+	 */
 	private $repositories;
 
 
 	/**
-	 * @param IRepository[] $repositories
+	 * @phpstan-param list<IRepository<\Nextras\Orm\Entity\IEntity>> $repositories
```
Comment (adaamz) [inline src/Model/SimpleRepositoryLoader.php:25]: Does phpstan know what is `list`? I think it needs to be `array<int, IRepository<\Nextras\Orm\Entity\IEntity>>`

Comment (adaamz) [reply src/Model/SimpleRepositoryLoader.php:25]: hm, I would better try it myself than write publicly 😄 
ok it works, thx for new info!

Comment (hrach) [reply src/Model/SimpleRepositoryLoader.php:25]: https://phpstan.org/r/6051a535-a402-42fa-af23-57ca3565630f

**Pull request merged**